xcode/Subconscious/Shared/Library/URLUtilities.swift

```diff
@@ -8,6 +8,37 @@
 import Foundation
 
 extension URL {
+    /// Attempt to construct a URL from the passed string and validate its components
+    /// This will only allow http & https URLs with a full hostname
+    init?(validatedString string: String) {
```

Comment:
Struggling with this name.

`validatedString` seems ambiguous to me, validated _for_ what or _as_ what etc.

Perhaps a better name is `httpUrlString:` to more specifically indicate usage?

Reply:
I like `URL(httpUrlString:)`.

More ideas:

```swift
URL(validating: String)
URL(strictlyValidating: String)
URL(validatingHttp: String)
URL(httpString: String)
```

Reply:
I think the tense should be present tense `validating`. `validated` suggests it has already been validated, to my eye.